plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestLdapAuthenticator.java

```diff
@@ -235,4 +238,69 @@ public void testContainsSpecialCharacters()
                 .as("Angle brackets")
                 .isEqualTo(true);
     }
+
+    @Test
+    public void testConnectTimeout()
+            throws Exception
+    {
+        try (DisposableSubContext organization = openLdapServer.createOrganization();
+                DisposableSubContext ignored = openLdapServer.createUser(organization, "alice", "alice-pass")) {
+            LdapConfig ldapConfig = new LdapConfig()
+                    .setLdapUrl(openLdapServer.getLdapUrl())
+                    .setLdapConnectTimeout(new Duration(1, MILLISECONDS))
```

Comment:
Is there a way to control how long the server gets to respond? Otherwise it looks like it's based on luck

Reply:
Actually we cannot configure the server to provide a timeout. 

> Otherwise it looks like it's based on luck
True, but ran it around 1k times it wasn't failing

Reply:
If we see flakiness due to this - we could use toxyproxy to increase the latency for queries.

Reply:
I would prefer preventing flakiness rather than fixing, though :)

Reply:
Well in case of connection timeout we can just create some artificial service will will open socket but will never respond. That will cause connection timeout for 100%.
Same for read timeout we just need to response with welcome package and afterwards nothing.

Reply:
@aczajkowski Have used ToxiProxy which adds latency for these tests thereby failing due to read and connect timeouts.

Reply:
Ok makes the job. Java server socket will be just more lightweight. But will be possibly hard to emulate ldap welcome response (no idea to be honest)